src/systemctl/systemctl-logind.c

```diff
@@ -96,16 +96,16 @@ int logind_check_inhibitors(enum action a) {
         char **s;
         int r;
 
-        if (arg_ignore_inhibitors || arg_force > 0)
+        if (arg_check_inhibitors == 0 || arg_force > 0)
                 return 0;
 
         if (arg_when > 0)
                 return 0;
 
-        if (geteuid() == 0)
+        if (geteuid() == 0 && arg_check_inhibitors <= 0)
                 return 0;
 
-        if (!on_tty())
+        if (!on_tty() && arg_check_inhibitors <= 0)
                 return 0;
```

Comment:
hmm, i'd write this as:

```c
if (arg_check_inhibitors < 0) {
        if (geteuid() == 0)
                return 0;
        if (!on_tty())
                return 0;
}
```

i.e. do the check only once, and only check `< 0` instead of `<= 0`, given that in the `== 0` case we shouldn't even get here, given the earlier check above, and if we'd cover that here again this would be quite confusing I think.